packages/features/insights/server/trpc-router.ts

```diff
@@ -1080,103 +1080,19 @@ export const insightsRouter = router({
       const { prisma, user } = ctx;
       const { teamId, userId, isAll } = input;
 
-      if (!teamId && !userId) {
-        return [];
-      }
-
-      const membershipWhereConditional: Prisma.MembershipWhereInput = {};
-
-      let childrenTeamIds: number[] = [];
-
-      if (isAll && teamId && user.organizationId && user.isOwnerAdminOfParentTeam) {
-        const childTeams = await prisma.team.findMany({
-          where: {
-            parentId: user.organizationId,
-          },
-          select: {
-            id: true,
-          },
-        });
-        if (childTeams.length > 0) {
-          childrenTeamIds = childTeams.map((team) => team.id);
-        }
-        membershipWhereConditional["teamId"] = {
-          in: [user.organizationId, ...childrenTeamIds],
-        };
-      }
-
-      if (teamId && !isAll) {
-        membershipWhereConditional["teamId"] = teamId;
-        membershipWhereConditional["userId"] = user.id;
-      }
-      if (userId) {
-        membershipWhereConditional["userId"] = userId;
-      }
-
-      // I'm not using unique here since when userId comes from input we should look for every
-      // event type that user owns
-      const membership = await prisma.membership.findFirst({
-        where: membershipWhereConditional,
-      });
-
-      if (!membership && !user.isOwnerAdminOfParentTeam) {
-        throw new Error("User is not part of a team/org");
-      }
-
-      const eventTypeWhereConditional: Prisma.EventTypeWhereInput = {};
-      if (isAll && childrenTeamIds.length > 0 && user.organizationId && user.isOwnerAdminOfParentTeam) {
-        eventTypeWhereConditional["teamId"] = {
-          in: [user.organizationId, ...childrenTeamIds],
-        };
-      }
-      if (teamId && !isAll) {
-        eventTypeWhereConditional["teamId"] = teamId;
-      }
-      if (userId) {
-        eventTypeWhereConditional["userId"] = userId;
-      }
-      let eventTypeResult: Prisma.EventTypeGetPayload<{
-        select: {
-          id: true;
-          slug: true;
-          teamId: true;
-          title: true;
-          team: {
-            select: {
-              name: true;
-            };
-          };
-        };
-      }>[] = [];
-
-      let isMember = membership?.role === "MEMBER";
-      if (user.isOwnerAdminOfParentTeam) {
-        isMember = false;
-      }
-      if (isMember) {
-        eventTypeWhereConditional["OR"] = [
-          { userId: user.id },
-          { users: { some: { id: user.id } } },
-          // @TODO this is not working as expected
-          // hosts: { some: { id: user.id } },
-        ];
-      }
-      eventTypeResult = await prisma.eventType.findMany({
-        select: {
-          id: true,
-          slug: true,
-          title: true,
-          teamId: true,
-          team: {
-            select: {
-              name: true,
-            },
-          },
+      const eventTypeList = await getEventTypeList({
```

Comment:
I've extracted this logic as a separate function, so that I can use it in a different function below.